EKF/mag_fusion.cpp

```diff
@@ -163,6 +163,15 @@ void Ekf::fuseMag()
 		return;
 	}
 
+	/*
+	When flying as a fixed wing aircraft, a misaligned magnetometer can cause an error in pitch/roll and accel bias estimates.
+	When MAG_FUSE_TYPE_AUTOFW is selected and the vehicle is flying as a fixed wing as detected by the use of wind estimation
+	and current airspeed estimates, then magnetometer fusion is only allowed to access the magnetic field states.
+	*/
+	_control_status.flags.update_mag_states_only = (_params.mag_fusion_type == MAG_FUSE_TYPE_AUTOFW)
+			&& _control_status.flags.wind
```

Comment:
why not use the _control_status.flags.fixed_wing instead?

Reply:
Good idea, I'll push a patch.